docs/user-guide/internal-database.rst

```diff
@@ -0,0 +1,39 @@
+*****************
+Internal Database
+*****************
+
+.. note:: This is a very advanced topic. Normal users should not need to read this page!
+
+The internal database of Cobbler is held at ``/var/lib/cobbler/collections``.
+
+.. note:: There is an option to use a MongoDB but this is untested at the moment. The content of each entry is the same
+          as when using the default serializer.
+
+Items
+#####
+
+An Item in Cobbler is a set of attributes grouped together and given a name. An example for this would be a ``distro``.
+On disk those items are represented using JSON. Per default the JSON is minified, however you can make the serializer
+produce "pretty" JSON files by changing ``serializer_pretty_json`` to ``true`` in the Cobbler Settings.
+
+The name of the saved file is the name of the item. However this will be subject to change in the future.
+
+Collections
+###########
+
+A collection in Cobbler is a number of ``n`` Cobbler Items that are living inside the same folder.
+
+Notes
+#####
+
+Currently items get deleted upon deletion inside Cobbler immediately. If you want to have a backup please use the
+``scm_track`` module of Cobbler. It will use Git for version control of the complete ``/var/lib/cobbler/`` folder.
+
+A rename operation is currently: Delete the item with the old name and create a new item with the new name. This is
+reflected on disk and thus if your Cobbler is being terminated at the right point in time, there is the extremely low
+probability that this specific item is lost. If you have items dependent onto that item you will receive errors on the
```

Comment:
I do want to document that we have this edge-case in the database. Under normal operation it is not even a bug because it is a valid approach to a rename. However one would not expect this behavior when facing this particular situation and thus the sentence. I will apply your suggestion.